File: bindgen-tests/swift/tests/custom_types.swift
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import uniffi_bindgen_tests

// CustomType1 doesn't have any Swift custom type associated with it
assert(roundtripCustomType1(custom1: 100) == 100)

// CustomType2 is a dict in swift.  See `uniffi.toml` for the configuration that does this.
assert(roundtripCustomType2(custom2: ["value": 200]) == ["value": 200])

File: bindgen-tests/ruby/tests/custom_types.rb
# frozen_string_literal: true

# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

require 'test/unit'
require 'uniffi_bindgen_tests'

class TestEnums < Test::Unit::TestCase
  include UniffiBindgenTests

  # CustomType1 has no foreign-side config: treat as raw u64
  def test_custom_type1
    assert_equal 100, UniffiBindgenTests.roundtrip_custom_type1(100)
  end

  # CustomType2 is lifted/lowered via a Ruby Hash per uniffi.toml Ruby config
  def test_custom_type2
    assert_equal({ 'value' => 200 }, UniffiBindgenTests.roundtrip_custom_type2({ 'value' => 200 }))
  end
end
